includes/rest/class-nodeinfo-controller.php

```diff
@@ -0,0 +1,215 @@
+<?php
+/**
+ * NodeInfo REST-Class file.
+ *
+ * @package Activitypub
+ */
+
+namespace Activitypub\Rest;
+
+use function Activitypub\get_total_users;
+use function Activitypub\get_active_users;
+use function Activitypub\get_rest_url_by_path;
+use function Activitypub\get_masked_wp_version;
+
+/**
+ * ActivityPub NodeInfo REST-Class.
+ *
+ * @author Matthias Pfefferle
+ *
+ * @see https://nodeinfo.diaspora.software/
+ */
+class Nodeinfo_Controller extends \WP_REST_Controller {
+	/**
+	 * The namespace of this controller's route.
+	 *
+	 * @var string
+	 */
+	protected $namespace = ACTIVITYPUB_REST_NAMESPACE;
+
+	/**
+	 * The REST base for this controller's route.
+	 *
+	 * @var string
+	 */
+	protected $rest_base = 'nodeinfo';
+
+	/**
+	 * Register routes.
+	 */
+	public function register_routes() {
+		\register_rest_route(
+			$this->namespace,
+			'/' . $this->rest_base,
+			array(
+				array(
+					'methods'             => \WP_REST_Server::READABLE,
+					'callback'            => array( $this, 'get_items' ),
+					'permission_callback' => '__return_true',
+				),
+			)
+		);
+
+		\register_rest_route(
+			$this->namespace,
+			'/' . $this->rest_base . '/(?P<version>\d\.\d)',
+			array(
+				'args' => array(
+					'version' => array(
+						'description' => 'The version of the NodeInfo schema.',
+						'type'        => 'string',
+						'enum'        => array( '2.0' ),
+						'required'    => true,
+					),
+				),
+				array(
+					'methods'             => \WP_REST_Server::READABLE,
+					'callback'            => array( $this, 'get_item' ),
+					'permission_callback' => '__return_true',
+				),
+			)
+		);
+
+		\register_rest_route(
+			$this->namespace,
+			'/' . $this->rest_base . '2',
+			array(
+				array(
+					'methods'             => \WP_REST_Server::READABLE,
+					'callback'            => array( $this, 'get_nodeinfo2' ),
+					'permission_callback' => '__return_true',
+				),
+			)
+		);
+	}
+
+	/**
+	 * Retrieves the NodeInfo discovery profile.
+	 *
+	 * @param \WP_REST_Request $request The request object.
+	 *
+	 * @return \WP_REST_Response Response object.
+	 */
+	public function get_items( $request ) { // phpcs:ignore VariableAnalysis.CodeAnalysis.VariableAnalysis.UnusedVariable
+		$response = array(
+			'links' => array(
+				array(
+					'rel'  => 'https://nodeinfo.diaspora.software/ns/schema/2.0',
+					'href' => get_rest_url_by_path( '/nodeinfo' ),
+				),
+				array(
+					'rel'  => 'https://www.w3.org/ns/activitystreams#Application',
+					'href' => get_rest_url_by_path( 'application' ),
+				),
+			),
+		);
+
+		return \rest_ensure_response( $response );
+	}
+
+	/**
+	 * Retrieves the NodeInfo profile.
+	 *
+	 * @param \WP_REST_Request $request The request object.
+	 * @return \WP_REST_Response Response object.
+	 */
+	public function get_item( $request ) {
```

Comment:
This is more consistent with the `get_nodeinfo2` function

```suggestion
	public function get_nodeinfo( $request ) {
```

Reply:
With the context of how `WP_REST_Controller ` is implemented, `get_nodeinfo2` is the inconsistent part, I'm afraid. I actually thought about suggesting to remove it, however I think it's used to populate https://the-federation.info?

Reply:
ah I see! ...this is a very "creative" way of using `get_items` and `get_item` in this context ☺️

the-federation.info also supports NodeInfo, so let's remove it then... there is a NodeInfo plugin that will bring back nodeinfo2 support if someone needs it... Keep it simple, stupid!

Reply:
Great! Removed in 2c304a3.